### DISCUSSION TITLE
Show first suggestion as "virtual text" on input line.

### DISCUSSION THREAD

**Discussion by TomJo2000** [marlonrichert/zsh-autocomplete#707] [Q&A]
Is there an existing way to show the first suggestion as "virtual text", similar to how `zsh-autosuggestions` does it
See example: 
![image](https://github.com/marlonrichert/zsh-autocomplete/assets/43716232/59041977-4649-41c8-9026-bf0b509be103)

I've combed through the documentation but I haven't found any way described to achieve this.
I'm pretty alright with shell scripting, but ZLE and zstyle elude me, so I'm unsure where to even begin here.

I did find the function responsible for doing this in [`zsh-autosuggestions`](https://github.com/zsh-users/zsh-autosuggestions/blob/c3d4e576c9c86eac62884bd47c01f6faed043fc5/zsh-autosuggestions.zsh#L242-L250).
```zsh
# If there was a highlight, remove it
_zsh_autosuggest_highlight_reset() {
    typeset -g _ZSH_AUTOSUGGEST_LAST_HIGHLIGHT


    if [[ -n "$_ZSH_AUTOSUGGEST_LAST_HIGHLIGHT" ]]; then
        region_highlight=("${(@)region_highlight:#$_ZSH_AUTOSUGGEST_LAST_HIGHLIGHT}")
        unset _ZSH_AUTOSUGGEST_LAST_HIGHLIGHT
    fi
}
```

However I assume it would have to be hooked to run after the fact in order to apply the styling properly.
It is unclear to me from the documentation and looking over the code if there is an existing way to hook this.

I'm really sorry to bother you with this, I fully realize this is - to use the colloqualism - a skill issue, on my part.

**Comment by marlonrichert** [comment]
You _should_ be able to get what you want with `ZSH_AUTOSUGGEST_STRATEGY=( completion )`, but that, for some reason, only completes the first word on the command line and nothing else. I would suggest you open an issue in the `zsh-autosuggestions` repo.
